.eslintrc.js

```diff
@@ -1,82 +1,98 @@
+    'react/prefer-stateless-function': 'error',
+    'react/prefer-stateless-function': [
+      'error',
+      { ignorePureComponents: true },
+    ],
+    'react/jsx-no-bind': 'error',
 
     // overriding recommended rules
-    "no-constant-condition": ["error", { checkLoops: false }],
-    "no-console": [ "error", { allow: ["log", "warn", "error"] } ],
-    "no-unused-vars": ["error", { "argsIgnorePattern": "^_" }],
+    'no-constant-condition': ['error', { checkLoops: false }],
+    'no-console': ['error', { allow: ['log', 'warn', 'error'] }],
+    'no-unused-vars': ['error', { argsIgnorePattern: '^_' }],
 
     // possible errors
-    "array-callback-return": "error",
-    "consistent-return": "error",
-    "default-case": "error",
-    "dot-notation": "error",
-    "eqeqeq": "error",
-    "for-direction": "error",
-    "no-alert": "error",
-    "no-caller": "error",
-    "no-eval": "error",
-    "no-extend-native": "error",
-    "no-extra-bind": "error",
-    "no-extra-label": "error",
-    "no-implied-eval": "error",
-    "no-invalid-this": "error",
-    "no-return-await": "error",
-    "no-self-compare": "error",
-    "no-throw-literal": "error",
-    "no-unmodified-loop-condition": "error",
-    "no-unused-expressions": "error",
-    "no-useless-call": "error",
-    "no-useless-computed-key": "error",
-    "no-useless-concat": "error",
-    "no-useless-constructor": "error",
-    "no-useless-rename": "error",
-    "no-useless-return": "error",
-    "no-var": "error",
-    "no-void": "error",
-    "no-with": "error",
-    "prefer-const": "error",
-    "prefer-promise-reject-errors": "error",
-    "prefer-rest-params": "error",
-    "prefer-spread": "error",
+    'array-callback-return': 'error',
+    'consistent-return': 'error',
+    'default-case': 'error',
+    'dot-notation': 'error',
+    eqeqeq: 'error',
+    'for-direction': 'error',
+    'no-alert': 'error',
+    'no-caller': 'error',
+    'no-eval': 'error',
+    'no-extend-native': 'error',
+    'no-extra-bind': 'error',
+    'no-extra-label': 'error',
+    'no-implied-eval': 'error',
+    // We use the version from the babel plugin so that `this` in a function
+    // class property doesn't give a false positive.
+    'babel/no-invalid-this': 'error',
+    'no-return-await': 'error',
+    'no-self-compare': 'error',
+    'no-throw-literal': 'error',
+    'no-unmodified-loop-condition': 'error',
+    'no-unused-expressions': 'error',
+    'no-useless-call': 'error',
+    'no-useless-computed-key': 'error',
+    'no-useless-concat': 'error',
+    'no-useless-constructor': 'error',
+    'no-useless-rename': 'error',
+    'no-useless-return': 'error',
+    'no-var': 'error',
+    'no-void': 'error',
+    'no-with': 'error',
+    'prefer-const': 'error',
+    'prefer-promise-reject-errors': 'error',
+    'prefer-rest-params': 'error',
+    'prefer-spread': 'error',
+  },
+  settings: {
+    react: {
+      pragma: 'React',
+      version: '15.0',
+      flowVersion: '0.54.1',
```

Comment:
specifying the flow version helps the react eslint plugin doing the right thing (I think that if we don't specify it it thinks it's the latest, which is OK for us because the syntax hasn't changed, but I'd rather specify it explicitely).

Reply:
Will this be hard to keep up to date?

Reply:
If it's not right we'll see it quickly because eslint will start complaining for absurd things. So I think we won't forget. It's OK if it's not exactly the right value.

Reply:
Here is what issue this can prevent:
* Flow releases a version where they changed how they type things for react
* we upgrade eslint-plugin-react but not Flow; that new version of eslint-plugin-react handles the new syntax, BUT thanks to this property it will still work for us until we upgrade Flow and update this property.

The case where we'll forget to update the property and has consequences is:
* Flow releases a version where they changed how they type things for react
* we upgrade Flow
* we upgrade eslint-plugin-react to a version that supports this new syntax, but forget to update this property
=> eslint will report absurd things

So I believe it's better to have it than not have it. It's not necessary _now_ but can help later.